app/loader.cpp

```diff
@@ -107,7 +108,13 @@ bool Loader::forceLoad( const QString &filePath, bool force )
     res = mProject->read( filePath );
     mActiveLayer.resetActiveLayer();
     mMapThemeModel.reloadMapThemes( mProject );
-    setActiveLayerByName( mAppSettings.defaultLayer() );
+
+    QgsVectorLayer *defaultLayer = mRecordingLayerPM.layerFromLayerName( mAppSettings.defaultLayer() );
+    if ( defaultLayer )
+      setActiveLayer( defaultLayer );
+    else
+      setActiveLayer( mRecordingLayerPM.firstUsableLayer() );
+
```

Comment:
Similar thing is happening here: https://github.com/lutraconsulting/input/blob/c8c8e23e584763ae36093f0ab33961a1ae628fbf/app/qml/main.qml#L211

Can we maybe erase one of these two?

Reply:
Similar, but I think a bit different. 
The former is called when a project is loaded and handles setting theme and layer accordingly.
The latter is used in `updateRecordToolbar` function which is called in several cases. Most relevant reason to keep it there is that it handles the case when:

1. user changes map theme which doesn't contain current active layer
2. starts to digitizing

Another option would be to replace/fix `Loader::setActiveLayerByName` used in `Loader::forceLoad`.
Essentially, the fix would be similar (checking if default layer is present in digitizing proxy layer model).